tests/src/core/testqgslayoutexporter.cpp

```diff
@@ -0,0 +1,260 @@
+  QCOMPARE( res, QgsLayoutExporter::Success );
+  QCOMPARE( layerIds, QList< unsigned int >() << 1 << 2 );
+  QCOMPARE( layerNames, QStringList() << QStringLiteral( "Pages" ) << QStringLiteral( "Labels, Shape" ) );
+  QCOMPARE( mapLayerIds, QStringList() << QString() << QString() );
+  layerIds.clear();
+  layerNames.clear();
+  mapLayerIds.clear();
+
+  // add an item which can only be used with other similar items, should break the next label into a different layer
+  QgsLayoutItemScaleBar *scaleBar = new QgsLayoutItemScaleBar( &l );
+  items << scaleBar;
+  res = exporter.handleLayeredExport( items, exportFunc );
+  QCOMPARE( res, QgsLayoutExporter::Success );
+  QCOMPARE( layerIds, QList< unsigned int >() << 1 << 2 << 3 );
+  QCOMPARE( layerNames, QStringList() << QStringLiteral( "Pages" ) << QStringLiteral( "Labels, Shape" ) << QStringLiteral( "<Scalebar>" ) );
+  QCOMPARE( mapLayerIds, QStringList() << QString() << QString() << QString() );
+  layerIds.clear();
+  layerNames.clear();
+  mapLayerIds.clear();
+
+  QgsLayoutItemLabel *label3 = new QgsLayoutItemLabel( &l );
+  items << label3;
+  res = exporter.handleLayeredExport( items, exportFunc );
+  QCOMPARE( res, QgsLayoutExporter::Success );
+  QCOMPARE( layerIds, QList< unsigned int >() << 1 << 2 << 3 << 4 );
+  QCOMPARE( layerNames, QStringList() << QStringLiteral( "Pages" ) << QStringLiteral( "Labels, Shape" ) << QStringLiteral( "<Scalebar>" ) << QStringLiteral( "<Label>" ) );
+  QCOMPARE( mapLayerIds, QStringList() << QString() << QString() << QString() << QString() );
+  layerIds.clear();
+  layerNames.clear();
+  mapLayerIds.clear();
+
+  // multiple scalebars can be placed in the same layer
+  QgsLayoutItemScaleBar *scaleBar2 = new QgsLayoutItemScaleBar( &l );
+  items << scaleBar2;
+  QgsLayoutItemScaleBar *scaleBar3 = new QgsLayoutItemScaleBar( &l );
+  items << scaleBar3;
+  res = exporter.handleLayeredExport( items, exportFunc );
+  QCOMPARE( res, QgsLayoutExporter::Success );
+  QCOMPARE( layerIds, QList< unsigned int >() << 1 << 2 << 3 << 4 << 5 );
+  QCOMPARE( layerNames, QStringList() << QStringLiteral( "Pages" ) << QStringLiteral( "Labels, Shape" ) << QStringLiteral( "<Scalebar>" ) << QStringLiteral( "<Label>" ) << QStringLiteral( "Scalebars" ) );
+  QCOMPARE( mapLayerIds, QStringList() << QString() << QString() << QString() << QString() << QString() );
+  layerIds.clear();
+  layerNames.clear();
+  mapLayerIds.clear();
+
+  // with an item which has sublayers
+  QgsVectorLayer *linesLayer = new QgsVectorLayer( TEST_DATA_DIR + QStringLiteral( "/lines.shp" ),
+      QStringLiteral( "lines" ), QStringLiteral( "ogr" ) );
+  QVERIFY( linesLayer->isValid() );
+  QgsVectorLayer *pointsLayer = new QgsVectorLayer( TEST_DATA_DIR + QStringLiteral( "/points.shp" ),
+      QStringLiteral( "points" ), QStringLiteral( "ogr" ) );
+  QVERIFY( pointsLayer->isValid() );
+
+  p.addMapLayer( linesLayer );
+  p.addMapLayer( pointsLayer );
```

Comment:
You are not using the pointsLayer anywhere after adding it to the project, did you mean to test multiple exported layer name IDs somewhere?

Reply:
Just a copy/paste error, I'll remove. Handling of multiple map layer's is tested in depth in testqgslayoutitemmap.cpp